chromatic_fitting.py

```diff
@@ -257,46 +418,27 @@ def attach_data(self, r):
         Rainbow object
         """
         self.data = r._create_copy()
-        self.apply_operation_to_constituent_models(LightcurveModel.attach_data, r)
+        self.apply_operation_to_constituent_models('attach_data', r)
+
+    def choose_optimization_method(self, optimization_method='simultaneous'):
+        LightcurveModel.choose_optimization_method(self, optimization_method)
+        if optimization_method == "separate":
+            self.apply_operation_to_constituent_models('change_all_priors_to_Wavelike')
 
     def setup_orbit(self):
         """
         Create an `exoplanet` orbit model, given the stored parameters.
         """
-        transit_model_exists = False
-        for mod in self.models_combined.values():
-            if isinstance(mod, TransitModel):
-                # if you set up the individual orbits then for whatever reason the parameters
-                # aren't fitted in the combined model?
-                # mod.setup_orbit()
-                transit_model_exists = True
-        if transit_model_exists:
-            TransitModel.setup_orbit(self)
+        self.apply_operation_to_constituent_models('setup_orbit')
 
     def setup_lightcurves(self):
         """
         Set-up lightcurves in combined model : for each consituent model set-up the lightcurves according to their type
         """
-
-        # I want to get this to work, but it doesn't seem to save the self.every_light_curve for both :(
-        # for mod in self.models_combined.values():
-        #     if isinstance(mod, TransitModel):
-        #         TransitModel.setup_lightcurves(self)
-        #     elif isinstance(mod, PolynomialModel):
-        #         PolynomialModel.setup_lightcurves(self)
-
-        data = self.get_data()
-        # WORKS BUT IS LESS NICE:
-        with self.model:
-            self.every_light_curve = {}
-            for i, w in enumerate(data.wavelength):
-                if f"wavelength_{i}" not in self.every_light_curve.keys():
-                    self.every_light_curve[f"wavelength_{i}"] = [0]*data.ntime
-                for mod in self.models_combined.values():
-                    if isinstance(mod, TransitModel):
-                        self.every_light_curve[f"wavelength_{i}"] += TransitModel.get_prior(self,i)
-                    elif isinstance(mod, PolynomialModel):
-                        self.every_light_curve[f"wavelength_{i}"] += PolynomialModel.get_prior(self,i)
+        self.every_light_curve = {}
+        self.apply_operation_to_constituent_models('setup_lightcurves')
+        for mod in self.chromatic_models.values():
+            self.every_light_curve = add_dicts(self.every_light_curve, mod.every_light_curve)
```

Comment:
Having this be just one function is really nice! It made it super easy to implement different operations in that `try-combined-model-operations` branch I experimented with.